### ISSUE TITLE
Editor backfill query is hardcoded to posts

### ISSUE THREAD

**Issue by dlh01** [alleyinteractive/wp-curate#54]
### Description of the bug

No matter the post types selected for the block, the `wp/v2/posts/` endpoint is used for searching, which returns only posts.

### Steps To Reproduce

1. Add `page` to the `wp_curate_duduplication_slotfill_post_types` filter.
2. Update the query block to draw backfill from only pages.
3. See posts display in the backfill instead of pages.

### Additional Information

See here: https://github.com/alleyinteractive/wp-curate/blob/b0d82733d8c0aebb698c16ff434c90ac742db71b/blocks/query/edit.tsx#L185.

### Acceptance Criteria

- Remove custom logic for posts endpoint https://github.com/alleyinteractive/wp-curate/blob/develop/src/features/class-rest-api.php
- Convert to using the search endpoint (if possible, even if the search endpoint needs to be augmented to include additional data for this type of query) or create a custom endpoint for WP Curate specifically.

**Comment by mogmarsh** [comment]
we probably need to create a custom endpoint or use `/wp/v2/search`

**Comment by mogmarsh** [comment]
ah yes, we had this fixed in the "first attempt": https://github.com/alleyinteractive/wp-curate/blob/first-attempt/src/class-wp-curate.php#L21-L35

**Comment by mogmarsh** [comment]
update: this is fixed, but I don't know if it was fixed in the best way.

**Comment by jacobschweitzer** [comment]
Working on re-factoring to use the `/search` endpoint.

**Comment by mogmarsh** [comment]
@jacobschweitzer I don't think `/search` will work, because we're using the core Post Template block, which references a post type endpoint. See https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-template/edit.js#L210
I think we decided the best fix is to create a dummy custom post type called `wp-curate` so that `/wp/v2/wp-curate` works and have it return posts from all the selected post types, rather than modifying the behavior of `wp/v2/posts/

**Comment by jacobschweitzer** [comment]
@mogmarsh I got a PR started though it needs a bit more work to get it across the finish line. I'm uncertain if this will be used in the project I'm on so I need to shift my focus back to other things for the time being.

**Comment by kevinfodness** [comment]
@mogmarsh is this still relevant?

**Comment by mogmarsh** [comment]
@kevinfodness yes and no... it works fine without it, but what we're doing feels a little "icky", so it's a nice to have, not a need to have at this point.
If we ever remove the dependency on the Post Template block (I guess by writing our own version of the block), this will become a non-issue, because we can use `/wp/v2/search` instead of `/wp/v2/posts`. If we keep the dependency on the Post Template block, then I still think the best fix is to create a dummy custom post type called `wp-curate` so that `/wp/v2/wp-curate` works and have it return posts from all the selected post types, rather than modifying the behavior of `/wp/v2/posts/`

**Comment by mogmarsh** [comment]
It looks like https://github.com/alleyinteractive/wp-curate/pull/251 may work, although it probably needs some updating.
